mitodl/mitxpro
# #1232 External course CMS page

#### Pre-Flight checklist


#### What are the relevant tickets?
#1232 

#### What's this PR do?
Adds a new CMS page type `External course page` that allows adding pages for courses that are hosted elsewhere. These pages also show up on the catalog page.

#### How should this be manually tested?
Under course index page, add an `ExternalCoursePage` type subpage. Fill in the required information and verify it appears in the catalog and on the detail page as expected.


#### Screenshots (if appropriate)
N.B.: Ignore the blank spaces in the screenshots, that's an artifact introduced by the screenshot tool. Doesn't appear on the pages.
![screencapture-xpro-odl-local-8053-courses-external-readable-id-2019-10-23-17_11_56](https://user-images.githubusercontent.com/45350418/67391730-b3275e00-f5b8-11e9-9f15-3883bf317e25.png)
![screencapture-xpro-odl-local-8053-courses-external-readable-id-2019-10-23-17_12_29](https://user-images.githubusercontent.com/45350418/67391735-b4f12180-f5b8-11e9-871e-d1a364da6ec0.png)
![screencapture-xpro-odl-local-8053-catalog-2019-10-23-17_13_55](https://user-images.githubusercontent.com/45350418/67391738-b6224e80-f5b8-11e9-98c2-987b25849e03.png)
![screencapture-xpro-odl-local-8053-catalog-2019-10-23-17_14_24](https://user-images.githubusercontent.com/45350418/67391744-b7537b80-f5b8-11e9-99e5-bb9b25b0cc90.png)
![Screenshot from 2019-10-28 15-26-03](https://user-images.githubusercontent.com/45350418/67671265-674d2e00-f997-11e9-89af-ea2cc9ded507.png)
![screencapture-xpro-odl-local-8053-cms-pages-add-cms-externalcoursepage-3-2019-10-28-15_26_15](https://user-images.githubusercontent.com/45350418/67671245-5dc3c600-f997-11e9-8a55-d078a6f73b50.png)
